pkg/customresource/customresource.go

```diff
@@ -157,10 +158,11 @@ func rawCRDFromFile(path string) ([]byte, error) {
 	return yamls.ReadFile(path)
 }
 
-func waitForCRDInit(context Context, resource CustomResource) error {
+func waitForCRDInit(crcontext Context, resource CustomResource) error {
 	crdName := fmt.Sprintf("%s.%s", resource.Plural, resource.Group)
-	return wait.Poll(context.Interval, context.Timeout, func() (bool, error) {
-		crd, err := context.APIExtensionClientset.ApiextensionsV1().CustomResourceDefinitions().Get(contextpkg.TODO(), crdName, metav1.GetOptions{})
+	deadlineCtx, _ := context.WithTimeout(context.TODO(), crcontext.Timeout)
+	return wait.PollUntilContextTimeout(deadlineCtx, crcontext.Interval, crcontext.Timeout, false, func(ctx context.Context) (bool, error) {
+		crd, err := crcontext.APIExtensionClientset.ApiextensionsV1().CustomResourceDefinitions().Get(contextpkg.TODO(), crdName, metav1.GetOptions{})
```

Comment:
We have used `contextpkg.TODO` and `context.TODO` both. What is the difference and why do we need both imports?

Reply:
Both are same. Thanks for pointing out. Addressed changes